julia/deps/build.jl

```diff
@@ -0,0 +1,261 @@
+import JSON
+
+################################################################################
+# First try to detect and load existing libmxnet
+################################################################################
+libmxnet_detected = false
+libmxnet_curr_ver = get(ENV, "MXNET_COMMIT", "master")
+curr_win = "20171019"  # v0.12.0
+
+if haskey(ENV, "MXNET_HOME")
+  MXNET_HOME = ENV["MXNET_HOME"]
+  info("MXNET_HOME environment detected: $MXNET_HOME")
+  info("Trying to load existing libmxnet...")
+  lib = Libdl.find_library("libmxnet.$(Libdl.dlext)",
+                           [joinpath(MXNET_HOME, "lib"), MXNET_HOME])
+  if !isempty(lib)
+    info("Existing libmxnet detected at $lib, skip building...")
+    libmxnet_detected = true
+  else
+    info("Failed to load existing libmxnet, trying to build from source...")
+  end
+end
+
+# Try to find cuda
+CUDAPATHS = String[]
+if haskey(ENV, "CUDA_HOME")
+  push!(CUDAPATHS, joinpath(ENV["CUDA_HOME"], "lib64"))
+elseif is_linux()
+  append!(CUDAPATHS, ["/opt/cuda/lib64", "/usr/local/cuda/lib64"])
+end
+
+if is_unix()
+  try
+    push!(CUDAPATHS, replace(strip(readstring(`which nvcc`)), "bin/nvcc", "lib64"))
+  end
+end
+
+HAS_CUDA = false
+HAS_CUDNN = false
+let cudalib = Libdl.find_library(["libcuda", "nvcuda.dll"], CUDAPATHS)
+  HAS_CUDA = !isempty(cudalib) && Libdl.dlopen_e(cudalib) != C_NULL
+end
+
+if !HAS_CUDA && is_windows()
+  # TODO: this needs to be improved.
+  try
+    run(`nvcc --version`)
+    HAS_CUDA = true
+  end
+end
+
+if HAS_CUDA  # then check cudnn
+  let cudnnlib = Libdl.find_library("libcudnn", CUDAPATHS)
+    HAS_CUDNN = !isempty(cudnnlib) && Libdl.dlopen_e(cudnnlib) != C_NULL
+    if HAS_CUDNN && !haskey(ENV, "CUDA_HOME")  # inference `CUDA_HOME`
+      ENV["CUDA_HOME"] = dirname(dirname(cudnnlib))
+    end
+  end
+end
+
+if HAS_CUDA
+  info("Found a CUDA installation.")
+  if HAS_CUDNN
+    info("Found a CuDNN installation.")
+  end
+  info("CUDA_HOME -> $(get(ENV, "CUDA_HOME", nothing))")
+else
+  info("Did not find a CUDA installation, using CPU-only version of MXNet.")
+end
+
+# propagate more build flags from ENV
+const CC  = get(ENV, "CC", nothing)
+const CXX = get(ENV, "CXX", nothing)
+const ADD_CFLAGS  = get(ENV, "ADD_CFLAGS", nothing)
+const ADD_LDFLAGS = get(ENV, "ADD_LDFLAGS", nothing)
+const USE_JEMALLOC = get(ENV, "USE_JEMALLOC", nothing)  # "0" or "1"
+
+function get_cpucore()
+    if haskey(ENV, "TRAVIS")  # on travis-ci
+        2
+    else
+        min(Sys.CPU_CORES, 32)
+    end
+end
+
+using BinDeps
+@BinDeps.setup
+if !libmxnet_detected
+  if is_windows()
+    if Sys.ARCH != :x86_64
+      info("Prebuilt windows binaries are only available on 64bit. You will have to built MXNet yourself.")
+      return
+    end
+    info("Downloading pre-built packages for Windows.")
+    base_url = "https://github.com/yajiedesign/mxnet/releases/download/weekly_binary_build_v2/prebuildbase_win10_x64_vc14_v2.7z"
```

Comment:
Migrate?

Reply:
Has url of the prebuilt binary for window been changed?

Reply:
Well we're trying to build off the target branch, right? How would this solution work with that requirement?

Reply:
Oh, I think end-user need it. 
For CI build, the env variable `MXNET_HOME` will override it.

Reply:
I see. I think we will have to revisit this code at a later point in stage as Julia will follow MXNets release cycles after the integration.